### PR TITLE
feat: implement MCP resource for list_sims

### LINKED ISSUE #78: [Feature]: Identify how we can utilise the broader Model Context Protocol
### Feature Description

I would like to explore how we can support the broader Model Context Protocol specification in addition to our existing tool and sampling capabilities.

I'm interested in how resources, prompts and elicitation could be used to improve the DX for the agent as well as the end-user. 

## Useful resources:
https://modelcontextprotocol.io/llms-full.txt
https://modelcontextprotocol.io/introduction
https://github.com/modelcontextprotocol/typescript-sdk

### Use Cases

_No response_

### Example Interactions

Resources could return content that is currently provided by discovery-based tools for example, `list_sims` provides data that is needed for tool calling.

### PR DESCRIPTION
**Pull request by cameroncooke**
Implements MCP resource support for `list_sims` as a targeted test

## Summary
- Add resources capability to MCP server configuration
- Create resource management system at `src/core/resources.ts`
- Implement `mcp://xcodebuild/simulators` resource URI
- Add comprehensive tests following no-vitest-mocking guidelines
- Maintain backward compatibility with existing `list_sims` tools
- Register resources in both static and dynamic modes

## Testing
- 15 comprehensive test cases covering success, error, and edge cases
- Follows project's no-vitest-mocking guidelines
- Uses dependency injection with `createMockExecutor`

Closes #78

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Added support for resource management, including a new interface for accessing iOS simulator data via resource URIs.
  * Resource registration is now integrated into server startup, enabling clients to discover and interact with available resources.
  * Server capabilities now advertise support for resource subscription and change notifications.

* **Tests**
  * Introduced comprehensive tests to verify resource export definitions, registration, and simulator data handling, including various success and error scenarios.

* **Documentation**
  * Updated testing guidelines to prohibit Vitest mocking patterns, enforcing dependency injection for all plugin handler tests.
  * Revised test examples and best practices to align with the new dependency injection strategy, improving test reliability and clarity.

### CODE REVIEW

**Comment by cameroncooke** [comment]
@claude Please address the concerns from these review comments, please look at `scripts/check-test-patterns.js` and `scripts/audit-dependency-container.js` which must be run to validate correct test patterns. Also update `docs/TESTING.md` as it's out-of-date with respect to mocking, the scripts and `CLAUDE.md` are correct with respect to mocking.

**Comment by cameroncooke** [comment]
@claude linting is failing causing checks to fail

**Comment by cameroncooke** [comment]
@claude tests are failing. You need to ensure you lint, format, build and test before committing.

**Comment by cameroncooke** [comment]
Tests are still failing use 'npm run tests'. @claude

**Comment by cameroncooke** [comment]
image

Still got a failing test. I'm unsure why you are having problems with test validation you said they all pass CI says otherwise. 

Is there a problem in running tests?

@claude

File: src/index.ts [hunk 1]
```diff
@@ -95,6 +102,9 @@ async function main(): Promise<void> {
           server.tool(plugin.name, plugin.description || '', plugin.schema, plugin.handler);
```
Comment (cameroncooke) [inline src/index.ts:102]: Why the ||

**Comment by cameroncooke** [comment]
bugbot run
